### PR TITLE
Adding DataObjectMixin inheritance to the `WellList` class. Also fixing an issue where adding and removing wells wasn't functioning as expected.

### PR DESCRIPTION
**Pull request by cflynn3**

### CODE REVIEW

File: ResSimpy/DataObjectMixin.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def __init__(self, _: dict[str, str | float | int | None] = {}, date: Optional[s
         self._date_format = date_format
         self._start_date = start_date
         self._unit_system = unit_system
-        self.__date = date
+        self.date = date
```
Comment (jonnyh404) [inline ResSimpy/DataObjectMixin.py:44]: shouldn't this be kept as _date?

File: ResSimpy/WellLists.py [hunk 2]
```diff
@@ -36,7 +26,7 @@ def welllists(self) -> Sequence[WellList]:
     @property
     def unique_names(self) -> list[str]:
         """Returns all WellList names."""
-        return list({x.name for x in self.welllists})
+        return list({x.name for x in self.welllists if x.name is not None})
```
Comment (jonnyh404) [inline ResSimpy/WellLists.py:29]: I think this should be self._well_lists

File: ResSimpy/Nexus/nexus_load_well_list.py [hunk 3]
```diff
@@ -45,14 +48,16 @@ def get_previous_well_list(well_list_name: str, well_lists: list[NexusWellList])
...
     """
-    for well_list in well_lists:
-        if well_list.name == well_list_name:
-            return well_list
-    return None
+
+    # Find the latest matching welllist
+    matching_well_lists = [x for x in well_lists if x.name == well_list_name]
+    sorted_well_lists = sorted(matching_well_lists, key=lambda x: x.iso_date)
+    return None if len(sorted_well_lists) == 0 else sorted_well_lists[-1]
```
Comment (jonnyh404) [inline ResSimpy/Nexus/nexus_load_well_list.py:55]: Good fix to existing code.

**Review by jonnyh404 — APPROVED** [review]
Looks good to me, please check comments for potential for a bug and private property.

File: ResSimpy/DataObjectMixin.py (hunk 1 above)
Comment (cflynn3) [reply ResSimpy/DataObjectMixin.py:44]: I changed this to instead kick off the property setter, that will then automatically set the iso_date.

File: ResSimpy/WellLists.py (hunk 2 above)
Comment (cflynn3) [reply ResSimpy/WellLists.py:29]: The public property of that name returns self._well_lists anyway, so I think this should be OK.

File: ResSimpy/WellList.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+
+@dataclass(kw_only=True)
+class WellList(DataObjectMixin):
+    """Class for representing a single WellList or group for the model."""
+    wells: list[str]
+
+    def __init__(self, date: str, wells: list[str], name: str, date_format: Optional[DateFormat] = None,
+                 start_date: Optional[str] = None) -> None:
+        """Initialises the OutputRequest class.
```
Comment (obi-isebor) [inline ResSimpy/WellList.py:15]: Shouldn't this be "Initialises the WellList class.?"

**Review by obi-isebor — APPROVED** [review]
Looks good! See minor docstring comment

**Pull request merged**